code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -310,16 +312,25 @@
 	return 1
 
 /obj/structure/closet/container_resist(mob/living/user)
-	if(opened || (!welded && !locked && !istype(loc, /obj/mecha)))
-		return  //Door's open, not locked or welded or inside a mech, no point in resisting.
+	if(opened)
+		return
+	if(istype(loc, /atom/movable))
+		user.changeNext_move(CLICK_CD_BREAKOUT)
+		user.last_special = world.time + CLICK_CD_BREAKOUT
+		var/atom/movable/AM = loc
+		AM.relay_container_resist(user, src)
+		return
+	if(!welded && !locked)
+		open()
+		return
 
-	//okay, so the closet is either welded or locked... resist!!!
+	//okay, so the closet is either welded or locked or inside something... resist!!!
 	user.changeNext_move(CLICK_CD_BREAKOUT)
 	user.last_special = world.time + CLICK_CD_BREAKOUT
-	user.visible_message("<span class='warning'>[src] begins to shake violently!</span>",
-						"<span class='notice'>You lean on the back of [src] and start pushing the door open.</span>")
+	user << "<span class='notice'>You lean on the back of [src] and start pushing the door open.</span>"
+	visible_message("<span class='warning'>[src] begins to shake violently!</span>")

```

Comment:
We have to give the visible_message to the closet, not the user, because otherwise noone but the user would get it.
I had to give the user his own message to inform them of their action, so they end up seeing two messages but I think it's worth it.